src/iOS/toga_iOS/widgets/base.py

```diff
@@ -78,6 +78,17 @@ def add_child(self, child):
         if self.container:
             child.viewport = self.root.viewport
             child.container = self.container
+        # The highest level box doesn't have a container - it is one
+        elif getattr(self, "viewport", None):
+            child.viewport = self.viewport
+            child.container = self
+
+    def remove_child(self, child):
+        # Remove the child UIView and all of its child subviews
+        child.native.removeFromSuperview()
+        for sub_child in child.interface.children:
+            if sub_child._impl:
+                child.remove_child(sub_child._impl)
```

Comment:
I think this can be simplified; the Cocoa backend has very similar mechanics, and pushes most of the "remove" logic to the setter for `container`.

Reply:
This was the way that I got it to work, if we don't call removeFromSuperview on all of the children (which this does indirectly) then the views don't seem to be removed from the view. But I'll have a look at the cocoa backend and see if this can be done less clunkily.

Reply:
hmm, I took russels advice to look at the cocoa implementation - I essentially copied the cocoa version and fixed what broke and now it works identical to the android version